tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -377,6 +377,11 @@ def stack_new_level_core(self, X, y, models: Union[List[AbstractModel], dict], X
                     for model_name in model_args_fit if 'hyperparameter_tune_kwargs' in model_args_fit[model_name]
                 }
                 kwargs['hyperparameter_tune_kwargs'] = hyperparameter_tune_kwargs
+                feature_prune_kwargs = {
```

Comment:
The way this is currently used is strange. I understand we originally did this because we were pruning on a per-model basis, but now it is very odd how it is leveraged. At minimum add code comment on why this is structured the way it is and why we give each model its own feature prune kwargs when we inevitably only train a single proxy model and ignore the other model's feature prune kwargs.